csrc/multidevice/pipeline.h

```diff
@@ -0,0 +1,142 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+#include <fusion.h>
+#include <ir/base_nodes.h>
+#include <multidevice/device_mesh.h>
+#include <multidevice/multidevice.h>
+#include <multidevice/pipeline_ir.h>
```

Comment:
Are these includes still needed?

Reply:
I removed `pipeline_ir.h` and added a forward declaration of `PipelineStage`